elytron/WFCORE-4150-credential-store-automatic-updates.adoc

```diff
@@ -0,0 +1,125 @@
+= WFCORE-4150 Support automatically adding and updating credentials in a credential store
+:author:            Farah Juma
+:email:             fjuma@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Many subsystems currently support a ```credential-reference``` which is used to specify either a reference to
+a credential that is stored in an Elytron credential store under a given alias or a clear text password. In
+particular, a ```credential-reference``` supports specifying either of the following two mutually exclusive
+things:
+
+1. ```store``` and ```alias``` attributes which specify the Elytron credential store and the alias within the
+store that hold a credential
+2. ```clear-text``` attribute which specifies a clear text password
+
+For the first case, a user needs to first create a credential store and add a credential to it before being able
+to reference it from a ```credential-reference```.
+
+This enhancement looks at relaxing the mutual exclusivity requirement to automate the process of adding or
+updating a credential in an existing credential store to simplify the number of steps that a user needs to
+perform.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4150[WFCORE-4150]
+* https://issues.jboss.org/browse/WFLY-12218[WFLY-12218]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-676[EAP7-676]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kasik]
```

Comment:
I am new QE contact for this RFE:

* mailto:tterem@redhat.com[Tomas Terem]

Reply:
Added.